### DISCUSSION TITLE
Python support for Visual Studio on ARM64

### DISCUSSION THREAD

**Discussion by vsfeedback** [Microsoft/PTVS#7788] [Enchancements]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/Python-support-for-Visual-Studio-on-ARM6/10233160)._

---
The ARM64 version of Visual Studio currently does not support Python development. Attempting to open .pyproj files will always result in an error.

Python 3.11 for Windows now officially supports ARM64 and there are native binaries available on python.org, so I assume adding support to Visual Studio shouldn't be too difficult.

---
#### Feedback Bot on 12/18/2022, 08:49 PM: 

(private comment, text removed)

**Comment by ghost** [comment]
Can agree that this is a feature that should be implemented

**Comment by gramster** [comment]
Right now our priority is getting the Python ecosystem itself bootstrapped properly on Windows ARM64. While we have ARM64 builds of CPython, many packages simply aren't available as wheels for ARM64 yet. We are focusing our resources on assisting that effort first.

**Comment by khmyznikov** [comment]
@gramster @judej  any news about this? Can't build scipy for example due lack of fortran compiler...

**Comment by cwebster-99** [comment]
Moving this issue to discussion as teams are still working on getting the Python ecosystem bootstrapped properly on Windows ARM64 before work can be done for PTVS support.

**Comment by Battant** [comment]
Hello,

I just test windows 11 pro for arm with vmware Fusion on wy macbook pro m1

It seem that python visual studo plugin is not avilabble on visual studio Community 2022 for windows arm

Is it possible that python will avilable in future version for arm ?

Thans for your comments

Best regards

**Comment by phisqb** [comment]
Hi there,

Any news on this or on an expected date for it? It's been almost a year and a half since the original post. You can't really use VS Pro 2022 very easily for python development on Surface Pro 11. If anyone has a simple workaround, it would be appreciated. All my environments and everything else were set up via VS Pro 2022 for multiple web applications. I was thinking of using WSL 2 as a temporary solution. Anyone else have any luck with something like that?

Thanks!

**Reply by Battant** [reply]
Hello,

My computer is a MacBook Pro I just test windows for ARM.

Because windows for arm should be develop in the futur for so windows support more architecture and visual studio is more accessible with touch screen and support zoom multitouch screan like IPad Pro with Microsoft RD client (com in soon windows app). I'm interested to test and use visual studio witch python for windows ARM on my MacBook Pro m1 in the futur.

Best regards

Battant

**Comment by phisqb** [comment]
Hi there,

Any update on this? Is there any other way to create a python project that will result in autocomplete, commenting shortcut keys, f12, and other basic project/solution features? Is there an ETA for when this might be completed? Currently, I'm using WSL but it's slow and the basic VS Pro functions mentioned above don't work without a proj/soln file which is super annoying.

Scott

**Comment by gramster** [comment]
We don’t have a clear date yet but it could be sometime in 2025. Right now
CPython on Windows ARM64 is still at tier 3 “experimental” support level
and numerous important packages don’t yet have binary wheels on PyPI.
We expect 2025 to be the year the various pieces come together.

**Comment by nohYoom** [comment]
Hello,

I hope you're all doing well. I wanted to ask if there have been any updates or progress regarding Python support for Visual Studio on ARM64. Any new information would be greatly appreciated. It would be nice to know the ETA.

Thank you for your time.

**Comment by ghost** [comment]
Please Microsoft, add Python support on arm64 Windows devices. Thank You!

**Comment by stucka** [comment]
There is now native support for ARM64 in Visual Studio, but there is no Python workload available: https://learn.microsoft.com/en-us/visualstudio/install/visual-studio-on-arm-devices?view=vs-2022

**Comment by gramster** [comment]
Python on Windows ARM64 is still 'experimental' at least until October 2026 with Python 3.15; see the announcement here: https://discuss.python.org/t/python-on-windows-arm64/104524

**Comment by arh-habib** [comment]
It's been nearly three years and Visual Studio 2026 is already released, yet it still doesn't support Python on ARM64-based PCs. With AI development becoming central to most solutions, native Python support on ARM64 is essential. Please prioritize adding full Python tooling and native runtime support for ARM64 to enable performant, modern AI workflows on these devices.

**Comment by phisqb** [comment]
Agreed....still waiting...
